db/crud.go

```diff
@@ -1762,7 +1764,7 @@ func (db *Database) updateAndReturnDoc(docid string, allowImport bool, expiry ui
 			}
 
 			docExists := currentValue != nil
-			syncFuncExpiry, newRevID, storedDoc, oldBodyJSON, unusedSequences, changedAccessPrincipals, changedRoleAccessUsers, err = db.documentUpdateFunc(docExists, doc, allowImport, docSequence, unusedSequences, callback, expiry)
+			syncFuncExpiry, newRevID, storedDoc, oldBodyJSON, unusedSequences, changedAccessPrincipals, changedRoleAccessUsers, avoidRevIDGeneration, err = db.documentUpdateFunc(docExists, doc, allowImport, docSequence, unusedSequences, callback, expiry)
```

Comment:
I don't think avoidRevIDGeneration makes sense as a variable name here - it sounds like revID generation is still pending, when in fact it's already been completed (or not completed) in documentUpdateFunc.  I think the only place it makes sense is inside documentUpdateFunc itself - for external usage, it's a flag that indicates whether a rev id was generated.